[cnv-4.18] port Automate vgpu vm upgrade test from cnv-tests repo

akri3i opened the pull request:
##### Short description:

##### More details:

##### What this PR does / why we need it:

##### Which issue(s) this PR fixes:

##### Special notes for reviewer:

##### jira-ticket:
<!--  full-ticket-url needs to be provided. This would add a link to the pull request to the jira and close it when the pull request is merged
If the task is not tracked by a Jira ticket, just write "NONE".
-->

akri3i commented:
/verified

rnetser reviewed (changes requested):
please close and once https://github.com/RedHatQE/openshift-virtualization-tests/pull/965, cherry-pick.
we should not have the same code being reviewed in multiple branches

hmeir commented:
@rnetser Actually, I would start with 4.18 and then cherry-pick to main.

Because the tests are marked with **pytest.mark.cnv**, cnv upgrade should run in order to verify this patch.

rnetser commented:
/retest tox
/retest build-container

akri3i commented:
> @rnetser Actually, I would start with 4.18 and then cherry-pick to main.
> 
> Because the tests are marked with **pytest.mark.cnv**, cnv upgrade should run in order to verify this patch.

@rnetser @hmeir , I've addressed all the reviews from [#965 ](url) here and closed it, will cherry-pick to main once this is merged

rnetser commented:
please fix tox

akri3i commented:
/verified

rnetser commented:
/approve

rnetser commented:
/lgtm

akri3i commented:
/cherry-pick main

openshift-virtualization-qe-bot-2 commented:
**Manual cherry-pick is needed**
Cherry pick failed for 4a3b28f3bbcf4ce76d635078e75c1d7646ae36b9 to main:
To cherry-pick run:
```
git remote update
git checkout main
git pull origin main
git checkout -b vgpu-upgrade-418-main
git cherry-pick 4a3b28f3bbcf4ce76d635078e75c1d7646ae36b9
git push origin vgpu-upgrade-418-main
```
